packages/syft/src/syft/client/syncing.py

```diff
@@ -193,6 +195,14 @@ def handle_sync_batch(
         )
         share_private_data_for_diff = share_private_data[diff.object_id]
         mockify_for_diff = mockify[diff.object_id]
+        if diff.object_type == "Job":
+            job = cast(Job, diff.get_obj())
+            if (
+                UID(job.result.id) not in share_private_data
+                or share_private_data[UID(job.result.id)] is False
+            ):
+                job.result = None
+                diff.set_obj(obj=job)
```

Comment:
Hi @abyesilyurt, I found that the problem lies here where in the `share_private_data` dict we have the job's value is True, but its result's value is False. Added a possibly temporary fix that set the `job.result` to None if this happens 
![Screenshot from 2024-07-17 15-46-41](https://github.com/user-attachments/assets/53d3e6ab-9e09-4ea6-a1ea-320edd3e0330)